contracts/Colony.sol

```diff
@@ -126,11 +128,34 @@ contract Colony is ColonyStorage {
   function getDomainCount() public view returns (uint256) {
     return domainCount;
   }
-
   function setFunctionReviewers(bytes4 _sig, uint8 _firstReviewer, uint8 _secondReviewer)
   private
   {
     uint8[2] memory _reviewers = [_firstReviewer, _secondReviewer];
     reviewers[_sig] = _reviewers;
   }
+
+  function verifyKey(bytes key) internal view returns (bool) {
+    uint256 colonyAddress;
+    uint256 skillid;
+    uint256 userAddress;
+    assembly {
+        colonyAddress := mload(add(key,32))
+        skillid := mload(add(key,52)) // Colony address was 20 bytes long, so add 20 bytes
+        userAddress := mload(add(key,84)) // Skillid was 32 bytes long, so add 32 bytes
+    }
+    colonyAddress >>= 96;
+    userAddress >>= 96;
+    // Require that the user is proving their own reputation in this colony.
+    require(address(colonyAddress) == address(this));
+    require(address(userAddress) == msg.sender);
+    return true;
+  }
+
+  function verifyReputationProof(bytes key, bytes value, uint branchMask, bytes32[] siblings) public view returns (bool) {  // solium-disable-line security/no-assign-params
+    require(verifyKey(key)==true);
```

Comment:
For me this is a modifier candidate but if you insist for this version just do `require(verifyKey(key));`